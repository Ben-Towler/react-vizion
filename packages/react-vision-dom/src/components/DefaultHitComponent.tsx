import React from 'react';
import { SearchResult } from './HitsComponent';

type Props = {
  searchResult?: SearchResult;
};

const DefaultHitComponent: React.FunctionComponent<Props> = ({
  searchResult,
}) => {
  return (
    <div className="default-hit-card">
      <div className="default-hit-container">
        <h4 className="default-hit-title">
          {searchResult && searchResult.resourceName}
        </h4>
        <div className="card-body">{JSON.stringify(searchResult, null, 2)}</div>
      </div>
    </div>
  );
};

export default DefaultHitComponent;
