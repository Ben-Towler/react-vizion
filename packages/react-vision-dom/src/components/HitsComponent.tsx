import React from 'react';
import DefaultHitComponent from './DefaultHitComponent';

export type SearchResult = {
  resourceId: string;
  address: any;
  distance: number;
  geoPoint: {
    lat: number;
    lng: number;
  };
  resourceName: string;
  note: string;
  onlineBookingUrl: string;
  phones: any;
  type: string;
};

type Props = {
  results: SearchResult[];
  className?: string;
  style?: React.CSSProperties;
  noResultsFound?: React.ReactNode;
  hit?: (result: SearchResult) => React.ReactNode;
};

const HitsComponent: React.FunctionComponent<Props> = ({
  results,
  className,
  style,
  hit,
  noResultsFound,
}) => {
  return (
    <div className={className} style={style}>
      {Array.isArray(results) && results.length > 0 ? (
        results.map(searchResult => (
          <div key={searchResult.resourceId}>
            {hit ? (
              hit(searchResult)
            ) : (
              <>
                <DefaultHitComponent searchResult={searchResult} />
              </>
            )}
          </div>
        ))
      ) : (
        <>
          {noResultsFound || (
            <div className="default-hit-no-results-found"></div>
          )}
        </>
      )}
      ;
    </div>
  );
};

export default HitsComponent;
