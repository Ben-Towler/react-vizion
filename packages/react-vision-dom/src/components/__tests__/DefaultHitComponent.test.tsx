import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import DefaultHitComponent from '../DefaultHitComponent';
import { SearchResult } from '../HitsComponent';

Enzyme.configure({ adapter: new Adapter() });

describe('DefaultHitComponent', () => {
  const searchResult: SearchResult = {
    resourceId: '1',
    resourceName: 'Test',
    note: 'test',
  };

  it('applies its default props', () => {
    const instance = renderer.create(
      //@ts-ignore
      <DefaultHitComponent refine={() => null} />
    );

    expect(instance.toJSON()).toMatchSnapshot();

    instance.unmount();
  });

  it('should render with the resource name in the title', () => {
    const wrapper = shallow(
      <DefaultHitComponent searchResult={searchResult} />
    );
    const titleElement = wrapper.find('.default-hit-title');

    expect(titleElement.text()).toEqual(searchResult.resourceName);
  });

  it('should render with the resource as a JSON in the card body', () => {
    const wrapper = shallow(
      <DefaultHitComponent searchResult={searchResult} />
    );
    const cardBodyElement = wrapper.find('.card-body');

    const expectedResult = JSON.stringify(searchResult, null, 2);

    expect(cardBodyElement.text()).toEqual(expectedResult);
  });
});
