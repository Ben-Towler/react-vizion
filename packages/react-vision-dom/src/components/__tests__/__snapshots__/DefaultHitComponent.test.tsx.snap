// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultHitComponent applies its default props 1`] = `
<div
  className="default-hit-card"
>
  <div
    className="default-hit-container"
  >
    <h4
      className="default-hit-title"
    />
    <div
      className="card-body"
    />
  </div>
</div>
`;
